File: src/main/java/com/hsmchurch/app/core/support/CrudStringFormat.java
package com.hsmchurch.app.core.support;

import java.util.function.Function;

public class CrudStringFormat {

    private static final String CREATE = "등록";
    private static final String READ = "조회";
    private static final String UPDATE = "업데이트";
    private static final String DELETE = "삭제";
    private static final String CANCEL = "취소";

    private static final Function<String, String> success = action -> String.format("%s 성공!", action);
    private static final Function<String, String> fail = action -> String.format("%s 실패!", action);
    private static final Base base = (entityName, actionResult) -> String.format("%s %s, request: {}", entityName, actionResult);
    private static final Base error = (entityName, actionResult) -> String.format("%s %s, request: {}, error: {}", entityName, actionResult);

    public static final Function<String, String> REGISTER_SUCCESS = entityName -> base.format(entityName, success.apply(CREATE));
    public static final Function<String, String> REGISTER_FAIL = entityName -> error.format(entityName, fail.apply(CREATE));
    public static final Function<String, String> READ_SUCCESS = entityName -> base.format(entityName, success.apply(READ));
    public static final Function<String, String> READ_FAIL = entityName -> error.format(entityName, fail.apply(READ));
    public static final Function<String, String> UPDATE_SUCCESS = entityName -> base.format(entityName, success.apply(UPDATE));
    public static final Function<String, String> UPDATE_FAIL = entityName -> error.format(entityName, fail.apply(UPDATE));
    public static final Function<String, String> DELETE_SUCCESS = entityName -> base.format(entityName, success.apply(DELETE));
    public static final Function<String, String> DELETE_FAIL = entityName -> error.format(entityName, fail.apply(DELETE));
    public static final Function<String, String> CANCEL_SUCCESS = entityName -> base.format(entityName, success.apply(CANCEL));
    public static final Function<String, String> CANCEL_FAIL = entityName -> error.format(entityName, fail.apply(CANCEL));

    @FunctionalInterface
    interface Base {
        String format(final String entityName,
                      final String actionResult);
    }

}

File: src/main/java/com/hsmchurch/app/common/support/CrudStringFormat.kt
package com.hsmchurch.app.common.support

object CrudStringFormat {

    private val CREATE = "등록"
    private val READ = "조회"
    private val UPDATE = "업데이트"
    private val DELETE = "삭제"
    private val CANCEL = "취소"

    private val success = { action: String -> "$action 성공!" }
    private val fail = { action: String -> "$action 실패!" }
    private val base = { entityName: String, actionResult: String -> "$entityName $actionResult, request: {}" }
    private val error = { entityName: String, actionResult: String -> "$entityName $actionResult, request: {}, error: {}" }

    val REGISTER_SUCCESS = { entityName: String -> base(entityName, success(CREATE)) }
    val REGISTER_FAIL = { entityName: String -> error(entityName, fail(CREATE)) }
    val READ_SUCCESS = { entityName: String  -> base(entityName, success(READ)) }
    val READ_FAIL = { entityName: String  -> error(entityName, fail(READ)) }
    val UPDATE_SUCCESS = { entityName: String  -> base(entityName, success(UPDATE)) }
    val UPDATE_FAIL = { entityName: String  -> error(entityName, fail(UPDATE)) }
    val DELETE_SUCCESS = { entityName: String  -> base(entityName, success(DELETE)) }
    val DELETE_FAIL = { entityName: String  -> error(entityName, fail(DELETE)) }
    val CANCEL_SUCCESS = { entityName: String  -> base(entityName, success(CANCEL)) }
    val CANCEL_FAIL = { entityName: String  -> error(entityName, fail(CANCEL)) }

}
